### PR TITLE
agent + query pipeline cleanups 

### PR DESCRIPTION
**Pull request by jerryjliu**
1) Added callbacks to the agent runner 

2) Cleaned up notebook:
With conditional links, it's a lot easier to express something like a ReAct agent in a query pipeline
The functions should be more modular, easier to read, etc.

### CODE REVIEW

File: docs/examples/agent/agent_runner/query_pipeline_agent.ipynb [hunk 1]
```diff
@@ -52,7 +52,7 @@
   },
```
Comment (hatianzhang) [inline docs/examples/agent/agent_runner/query_pipeline_agent.ipynb:52]: <p>it seems it is not showing the graph tho</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/10563/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (jerryjliu) [reply docs/examples/agent/agent_runner/query_pipeline_agent.ipynb:52]: good point, need to clear it

File: llama_index/agent/runner/base.py [hunk 2]
```diff
@@ -208,15 +208,33 @@ def __init__(
...
+        verbose: bool = False,
     ) -> None:
         """Initialize."""
         self.agent_worker = agent_worker
         self.state = state or AgentState()
         self.memory = memory or ChatMemoryBuffer.from_defaults(chat_history, llm=llm)
-        self.callback_manager = callback_manager or CallbackManager([])
+
+        # get and set callback manager
+        if callback_manager is not None:
```
Comment (logan-markewich) [inline llama_index/agent/runner/base.py:219]: we should reeealllly just make the callback manager global. This is possible with v0.10.0

Theres a ton of brittle callback manager inheritance going on lol

Comment (jerryjliu) [reply llama_index/agent/runner/base.py:219]: true. will take a deeper look when i make the agents video

**Pull request merged**